### ISSUE TITLE
Lazy DataScollerRenderer loads data twice

### ISSUE THREAD

**Issue by nunofaria11** [primefaces/primefaces#390]
The current **lazy** `DataScrollerRenderer` implementation loads data twice in its `encodeMarkup` method. Suggested correction:

```
public class DataScrollerRenderer extends CoreRenderer {

    protected void encodeMarkup(FacesContext context, DataScroller ds, int chunkSize) throws IOException {
        // ... 
        boolean alreadyLoaded = false;
        if (ds.isLazy()) {            
            alreadyLoaded = true;
            loadLazyData(ds, 0, chunkSize);
        }
        // ... 
        loadChunk(context, ds, 0, chunkSize, alreadyLoaded);
        // ... 
    }

    protected void loadChunk(FacesContext context, DataScroller ds, int start, int size) throws IOException {
        loadChunk(context, ds, start, size, false);
    }

    private void loadChunk(FacesContext context, DataScroller ds, int start, int size, boolean alreadyLoaded) throws IOException {
        // ... 
        if (ds.isLazy() && !alreadyLoaded) {
            loadLazyData(ds, start, size);
        }
        // ... 
    }

}
```

SO post: http://stackoverflow.com/q/29303228/691916

**Comment by kukel** [comment]
Duplicate: https://github.com/primefaces/primefaces/issues/290? Read that issue up to the end

**Comment by nunofaria11** [comment]
Yes this is a duplicate. Thanks

**Comment by nunofaria11** [comment]
Actually after taking a look at it, I believe it is solely a `DataScrollerRenderer` issue, not a `LazyDataModel` issue. If you take a look at (debug) the current implementation, when rendering a lazy `DataScroller` the data model loads data twice for the same _page_ (0 to `chunkSize`). 

So I reopened this issue - the suggested correction works for me.

**Comment by cagataycivici** [comment]
Set to review.

**Comment by Dsleeper** [comment]
Is this confirmed as fixed? I am still having this issue with 5.3.RC2

**Comment by rider87** [comment]
Can you review this bug, please? Some users still have the issue, see also in the PF forum.
